graphql/ratelimit.go

```diff
@@ -0,0 +1,172 @@
+package graphql
+
+import (
+	"sync"
+	"time"
+
+	"github.com/satori/go.uuid"
+)
+
+type endRequestState int
+
+const (
+	maxSimultaneousRequestsDefault int             = 15
+	minSimultaneousRequestsDefault int             = 2
+	waitTimeDefault                time.Duration   = 3 * time.Second
+	endRequestStateOK              endRequestState = 0
+	endRequestStateError           endRequestState = 1
+	endRequestStateTimedOut        endRequestState = 2
+)
+
+// RatelimitObject represents structure of ratelimit object
+type RatelimitObject struct {
+	// requestsBucket a list of all simultaneous requests
+	requestsBucket []activeRequest
```

Comment:
I see two problems with `requestsBucket`:
1. A `map[uuid]request` would work better here (easier to remove elements).
2. It is not actually used for anything! We just add queries in the bucket when they start, and remove when they finish.